### PR TITLE
add experimental jax.no_tracing context manager

### PR DESCRIPTION
**Pull request by mattjj**
I don't _think_ we need to change `trace_context` (or any of the other contexts).

### CODE REVIEW

File: jax/_src/config.py [hunk 1]
```diff
@@ -1501,6 +1501,11 @@ def _update_disable_jit_thread_local(val):
     upgrade=True,
     help='Enable eager-mode pmap when jax_disable_jit is activated.')
 
+no_tracing = bool_state(
+    name='jax_no_tracing',
+    default=False,
+    help='Disallow tracing for JIT compilation.')
```
Comment (yashk2810) [inline jax/_src/config.py:1507]: You might wanna add (after it's traced for the first time?)

**Review by yashk2810 — APPROVED** [review]
This solution is perfect!

**Comment by rwitten** [comment]
Amazing thank you!

File: jax/_src/config.py (hunk 1 above)
Comment (jekbradbury) [reply jax/_src/config.py:1507]: /I think this should (and does?) trigger even if it's the first time

**Comment by saeta** [comment]
Thank you @mattjj !!!

**Pull request merged**